slider/beatmap.py

```diff
@@ -1414,31 +1415,230 @@ def ar(self,
+            enabled.
+        double_time : bool, optional
+            Get the effective position of the hit objects with double time
+            enabled.
 
-    @lazyval
-    def sliders(self):
-        """Just the sliders in the beatmap.
+        Returns
+        -------
+        hit_objects : list[HitObject]
+            The objects with their effective positions and timings given the
+             parameter set.
         """
-        return tuple(e for e in self.hit_objects if isinstance(e, Slider))
+
+        hit_objects = self._hit_objects
+
+        if hard_rock:
+            hit_objects = [ob.hard_rock for ob in hit_objects]
+
+        if stacking:
+            ar = self.ar(easy=easy, hard_rock=hard_rock)
+            cs = self.cs(easy=easy, hard_rock=hard_rock)
+            print(self.format_version)
+            if self.format_version >= 6:
+                hit_objects = self._resolve_stacking(hit_objects, ar, cs)
+            else:
+                hit_objects = self._resolve_stacking_old(hit_objects, ar, cs)
+
+        if double_time:
+            hit_objects = [ob.double_time for ob in hit_objects]
+        elif half_time:
+            hit_objects = [ob.half_time for ob in hit_objects]
+
+        if not spinners:
+            hit_objects = tuple(
+                ob for ob in hit_objects if not isinstance(ob, Spinner))
+
+        if not circles:
+            hit_objects = tuple(
+                ob for ob in hit_objects if not isinstance(ob, Circle))
+
+        if not sliders:
+            hit_objects = tuple(
+                ob for ob in hit_objects if not isinstance(ob, Slider))
+
+        return hit_objects
+
+    def _resolve_stacking(self, hit_objects, ar, cs):
+        stack_threshold = ar_to_ms(ar) * self.stack_leniency
+        stack_threshold = timedelta(milliseconds=stack_threshold)
+        stack_dist = 3
+        stack_height = {ob: 0 for ob in hit_objects}
+        # reverse list so it's easier to process
+        hit_objects = list(reversed(hit_objects))
+
+        for i, ob_i in enumerate(hit_objects):
+
+            if stack_height[ob_i] != 0 or isinstance(ob_i, Spinner):
+                continue
+
+            if isinstance(ob_i, Circle):
+                for n, ob_n in enumerate(hit_objects[i:], start=i):
+
+                    if isinstance(ob_n, Spinner):
+                        continue
+
+                    if hasattr(ob_n, "end_time"):
+                        end_time = ob_n.end_time
+                    else:
+                        end_time = ob_n.time
+
+                    if (ob_i.time - end_time) > stack_threshold:
+                        break
+
+                    if (isinstance(ob_n, Slider) and
+                            distance(ob_n.curve(1),
+                                     ob_i.position) < stack_dist):
+                        offset = stack_height[ob_i] - stack_height[ob_n] + 1
+
+                        for j, hj in enumerate(hit_objects[n:], start=n):
```

Comment:
enumerate is going the wrong direction
```suggestion
                        for j, hj in enumerate(hit_objects[i:n], start=i):
```